lib/sparql_to_sw_solr/instance_topic_fields.rb

```diff
@@ -0,0 +1,38 @@
+module SparqlToSwSolr
+  class InstanceSolrDoc
+    module InstanceTopicFields
+
+      private
+
+      def add_doc_topic_fields(doc)
+        doc[:topic_search] = topics
+        doc[:topic_facet] = topics
+        doc
+      end
+
+      def topics
+        @topics ||= begin
+          results = sparql_query topic_query
+          results.map { |t| topic_parse t[:topicLabel] }
+        end
+      end
+
+      def topic_parse(topic)
+        t = topic.to_s.split('--').first
```

Comment:
if you can't grab me for a conversation about subject field values, I recommend you speak to Arcadia or Nancy.  The format of  "broad--narrower--narroweryet" etc as a subject string is common and represents a single value.

I also did a quick scan thru the topic facet values at https://sw-webapp-sandbox-e.stanford.edu/?f%5Baccess_facet%5D%5B%5D=bibframe and found values like `Conductors (Music`  which are clearly parsed incorrectly.  (Welcome to library data.)  The trailing punctuation to be removed is more specific.  So for the facet field, `Conductors (Music)--Italy--Biography.` becomes `Conductors (Music)` per the spec in GoogleDocs.

As an example:  search on the ckey in production and on our test sw instance.  For example: https://searchworks.stanford.edu/?utf8=%E2%9C%93&search_field=search&q=10415005  vs  https://sw-webapp-sandbox-e.stanford.edu/?q=10415005&search_field=search.  with our marc data, the "Italy" part is in a particular subfield that we index as the "Region" facet;  the notes in https://docs.google.com/document/d/1Dlnfxey_gdhBq8Dnef8kyRw8mMpIE95r5AC2APi0Dgs  seem to indicate that we can't split the Region stuff out any more.   

To look at the admittedly yucky java code (which is improved over the even yuckier community version of the code I started from), check out
- https://github.com/sul-dlss/solrmarc-sw/blob/master/stanford-sw/sw_index.properties#L87 which indicates the marc fields and subfields and leads to 
- https://github.com/sul-dlss/solrmarc-sw/blob/b480166ad45d83762c3f4415100d9232a3c071ec/stanford-sw/src/edu/stanford/StanfordIndexer.java#L1010-L1040

which essentially indicates the trailing punctuation to be removed is `\`, `,`  `:`, `;` and periods, sometimes (don't worry about them for now)